### PR TITLE
Spring 5

### PR DESCRIPTION
**Pull request by ericenns**
## Description of changes
Upgraded Spring and Spring Security to latest 5.X releases.

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/RESTGenericController.java [hunk 1]
```diff
@@ -106,7 +104,6 @@ protected RESTGenericController(CRUDService<Long, Type> crudService, Class<Type>
 	 * @return a model containing all resources of the specified type in the
 	 *         application.
 	 */
-	@RequestMapping(method = RequestMethod.GET)
```
[the remark this answers is not in the document]
Comment (ericenns) [reply src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/RESTGenericController.java]: I was just following what you did [here](https://github.com/tom114/irida/commit/4cbaf784a8d775cc6159ffdd19e94d7bfc499682)

File: src/test/java/ca/corefacility/bioinformatics/irida/web/controller/test/integration/sequencingrun/SequencingRunIT.java [hunk 2]
```diff
@@ -89,7 +89,7 @@ public void testCreateRunAsSequencerSucceed() {
 	@Test
 	public void testPostSequencingRunFail() {
 		Map<String, String> run = createRun();
-		asSequencer().given().body(run).expect().response().statusCode(HttpStatus.SC_BAD_REQUEST).when()
+		asSequencer().given().body(run).expect().response().statusCode(HttpStatus.SC_METHOD_NOT_ALLOWED).when()
```
[the remark this answers is not in the document]
Comment (ericenns) [reply src/test/java/ca/corefacility/bioinformatics/irida/web/controller/test/integration/sequencingrun/SequencingRunIT.java]: This is the same change from when you attempted to upgrade to spring 5, see [here](https://github.com/tom114/irida/commit/3d4f9456d83aa3bf9bcb1931fa50560b0bb39dc9)

**Comment by ericenns** [comment]
> Build failing on my machine and Travis:
> 
> ```
> [INFO] -------------------------------------------------------------
> [ERROR] COMPILATION ERROR : 
> [INFO] -------------------------------------------------------------
> [ERROR] /home/travis/build/phac-nml/irida/src/main/java/ca/corefacility/bioinformatics/irida/config/security/IridaWebSecurityConfig.java:[53,29] configure(org.springframework.security.config.annotation.web.builders.WebSecurity) in ca.corefacility.bioinformatics.irida.config.security.IridaWebSecurityConfig.UISecurityConfig cannot override configure(org.springframework.security.config.annotation.web.builders.WebSecurity) in org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
>   overridden method does not throw java.lang.Exception
> ```

Yeah, I ended up downgrading to 5.1.x. There are more issues then just this one with the upgrade to 5.2.X

File: src/test/java/ca/corefacility/bioinformatics/irida/web/controller/test/integration/sequencingrun/SequencingRunIT.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tom114) [reply src/test/java/ca/corefacility/bioinformatics/irida/web/controller/test/integration/sequencingrun/SequencingRunIT.java]: Actually yeah scratch that, in this case it should be a method not allowed response here.  Resolving.

File: src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/sequencingrun/RESTSequencingRunController.java [hunk 3]
```diff
@@ -57,10 +62,23 @@ public RESTSequencingRunController(SequencingRunService service) {
...
+	 * Exception Handler for {InvalidDefinitionException} this basically
+	 * prevents create endpoint for SequencingRun from existing. Since
+	 * SequencingRun is an Abstract class it does not have a valid deserializer
+	 * so jackson throws a InvalidDefinitionException. We handle that exception
+	 * and return a status code of 405 (Method not allowed).
+	 */
+	@ExceptionHandler(InvalidDefinitionException.class)
+	@ResponseStatus(HttpStatus.METHOD_NOT_ALLOWED)
+	public void handleInvalidDefinitionException() {
+		//
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/sequencingrun/RESTSequencingRunController.java]: Ditch the empty comment please

File: src/main/java/ca/corefacility/bioinformatics/irida/service/impl/analysis/submission/AnalysisSubmissionServiceImpl.java [hunk 4]
```diff
@@ -388,31 +383,16 @@ public AnalysisSubmission updatePriority(AnalysisSubmission submission, Analysis
...
-						&& primaryCause.getCause() instanceof TransientPropertyValueException) {
-					final TransientPropertyValueException propertyException = (TransientPropertyValueException) primaryCause
-							.getCause();
-					if (Objects.equals("namedParameters", propertyException.getPropertyName())) {
-						throw new UnsupportedOperationException(
-								"You must save the named properties *before* you use them in a submission.", e);
-					}
-				}
-			}
+		return super.create(analysisSubmission);
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/service/impl/analysis/submission/AnalysisSubmissionServiceImpl.java]: I ditched the exception wrapping here.  We were just changing the `InvalidDataAccessApiUsageException` to a `UnsupportedOperationException`.   It was no longer getting thrown on `super.create(analysisSubmission)`, but instead when the transaction fired at the end of the method.  We weren't ever catching the exception anywhere except in the corresponding test so no issue with refactoring it out.

File: src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/sequencingrun/RESTSequencingRunController.java (hunk 3 above)
Comment (ericenns) [reply src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/sequencingrun/RESTSequencingRunController.java]: Fixed in dac4d2a

**Pull request merged**